File: examples/nodejs/submit_order_with_attached.js
const {
  Config,
  TradeContext,
  Decimal,
  OrderSide,
  TimeInForceType,
  OrderType,
  AttachedOrderType,
  OAuth,
} = require('longbridge');

async function main() {
  const oauth = await OAuth.build("your-client-id", (_, url) => {
    console.log("Open this URL to authorize: " + url);
  });
  let config = Config.fromOAuth(oauth);
  let ctx = TradeContext.new(config);

  let resp = await ctx.submitOrder({
    symbol: "AAPL.US",
    orderType: OrderType.LO,
    side: OrderSide.Buy,
    submittedQuantity: new Decimal(1),
    timeInForce: TimeInForceType.Day,
    submittedPrice: new Decimal(200),
    attachedParams: {
      attachedOrderType: AttachedOrderType.Bracket,
      profitTakerPrice: new Decimal(220),
      stopLossPrice: new Decimal(180),
    },
  });
  console.log(resp.toString());
}

Promise.all([main()]).catch((err) => console.error(err));

File: examples/python/submit_order_with_attached.py
from decimal import Decimal

from longbridge.openapi import (
    AttachedOrderType,
    Config,
    OAuthBuilder,
    OrderSide,
    OrderType,
    SubmitAttachedParams,
    TimeInForceType,
    TradeContext,
)

oauth = OAuthBuilder("your-client-id").build(
    lambda url: print(f"Open this URL to authorize: {url}")
)
config = Config.from_oauth(oauth)
ctx = TradeContext(config)

resp = ctx.submit_order(
    symbol="AAPL.US",
    order_type=OrderType.LO,
    side=OrderSide.Buy,
    submitted_quantity=Decimal(1),
    time_in_force=TimeInForceType.Day,
    submitted_price=Decimal(200),
    attached_params=SubmitAttachedParams(
        AttachedOrderType.Bracket,
        profit_taker_price=Decimal(220),
        stop_loss_price=Decimal(180),
    ),
)
print(resp)
